packages/kit/src/DropdownButton.tsx

```diff
@@ -1,70 +1,59 @@
-import React from 'react'
-import { Button, Sheet, YStack } from 'tamagui'
-import { z } from 'zod'
+/** @jsxImportSource @final-ui/react */
 
-const DropdownButtonProps = z.object({
-  id: z.string().optional(),
-  onSelect: z.function().args(z.string()).optional(),
-  buttonProps: z.any().optional(),
-  options: z
-    .array(
-      z.object({
-        key: z.string(),
-        label: z.string(),
-        icon: z.any().optional(),
-      })
-    )
-    .optional(),
-})
+import { extend, Only, setStateAction, state, WithServerProps } from '@final-ui/react'
+import {
+  Button,
+  Sheet,
+  SheetFrame,
+  SheetHandle,
+  SheetOverlay,
+  YStack,
+} from '@final-ui/tamagui/server'
 
-export function DropdownButton(props: z.infer<typeof DropdownButtonProps>) {
-  const { options } = props
-  const [open, setOpen] = React.useState(false)
+type Props = {
+  onSelect?: (item: string) => void
+  button?: JSX.Element
+  options?: {
+    key: Only<string>
```

Comment:
We automatically extend each `string` to also accept `StateDataState` since we can resolve state on the client down to primitive value.

However, `key` is special and there might be other special properties in the future that do not require/should not be anything but primitive value.

On one hand, we could consider adding support for `key` to be a `StateDataState` (in `template.tsx`, wrap `stateNode.key` with `renderProp` function call), but I find this out of scope.

This little helper is just to annotate type and is used within `WithServerProps` utility to detect which types should be extended.